src/fu-engine.c

```diff
@@ -1001,6 +1002,8 @@ fu_engine_modify_bios_settings(FuEngine *self,
 				    (const gchar *)key);
 			return FALSE;
 		}
+		if (g_strcmp0(key, FWUPD_BIOS_SETTING_SELF_TEST) == 0)
+			return TRUE;
```

Comment:
I still think it's better to pass a directory of real files to the daemon for installed tests; it will run more code.

Reply:
The problem with this approach is that it's perfectly acceptable to run the actual daemon and then run the installed tests against *that* -- I don't think it makes sense to say "switch to these bios settings" over DBus. I guess we could conditionalize on env("CI") but it feels really fragile.

Reply:
I got another idea.  How about when we enable test devices we also use moar test specific stuff like our own tmpfs directory?

Reply:
It doesn't give me good vibes; we use the DMI data for the vendor ID and although that shouldn't be affected I think it's a bit unexpected that we change the exposed BIOS settings if we're just enabling some test devices. I _think_ it makes more sense to check this code from the self tests, rather than emulating end-to-end over dbus.